api/v3/rpc.go

```diff
@@ -118,79 +114,41 @@ func (s *AncestryServer) GetAncestry(ctx context.Context, req *pb.GetAncestryReq
 	if err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
+
 	defer tx.Rollback()
 
-	if req.GetWithFeatures() || req.GetWithVulnerabilities() {
-		ancestry, ok, err := tx.FindAncestryWithContent(name)
-		if err != nil {
-			return nil, status.Error(codes.Internal, err.Error())
-		}
+	ancestry, ok, err := tx.FindAncestry(name)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
 
-		if !ok {
-			return nil, status.Error(codes.NotFound, fmt.Sprintf("requested ancestry '%s' is not found", req.GetAncestryName()))
-		}
+	if !ok {
+		return nil, status.Error(codes.NotFound, fmt.Sprintf("requested ancestry '%s' is not found", req.GetAncestryName()))
+	}
 
-		respAncestry = &pb.GetAncestryResponse_Ancestry{
-			Name:             name,
-			ScannedDetectors: ancestry.ProcessedBy.Detectors,
-			ScannedListers:   ancestry.ProcessedBy.Listers,
-		}
+	pbAncestry := &pb.GetAncestryResponse_Ancestry{
+		Name:             ancestry.Name,
+		ScannedDetectors: ancestry.ProcessedBy.Detectors,
+		ScannedListers:   ancestry.ProcessedBy.Listers,
+	}
 
-		for _, layer := range ancestry.Layers {
-			ancestryLayer := &pb.GetAncestryResponse_AncestryLayer{
-				Layer: &pb.Layer{
-					Hash: layer.Hash,
-				},
-			}
-
-			if req.GetWithVulnerabilities() {
-				featureVulnerabilities, err := tx.FindAffectedNamespacedFeatures(layer.DetectedFeatures)
-				if err != nil {
-					return nil, status.Error(codes.Internal, err.Error())
-				}
-
-				for _, fv := range featureVulnerabilities {
-					// Ensure that every feature can be found.
-					if !fv.Valid {
-						return nil, status.Error(codes.Internal, "ancestry feature is not found")
-					}
-
-					feature := pb.NamespacedFeatureFromDatabaseModel(fv.NamespacedFeature)
-					for _, v := range fv.AffectedBy {
-						vuln, err := pb.VulnerabilityWithFixedInFromDatabaseModel(v)
-						if err != nil {
-							return nil, status.Error(codes.Internal, err.Error())
-						}
-						feature.Vulnerabilities = append(feature.Vulnerabilities, vuln)
-					}
-					ancestryLayer.DetectedFeatures = append(ancestryLayer.DetectedFeatures, feature)
-				}
-			} else {
-				for _, dbFeature := range layer.DetectedFeatures {
-					ancestryLayer.DetectedFeatures = append(ancestryLayer.DetectedFeatures, pb.NamespacedFeatureFromDatabaseModel(dbFeature))
-				}
-			}
-
-			respAncestry.Layers = append(respAncestry.Layers, ancestryLayer)
-		}
-	} else {
-		dbAncestry, ok, err := tx.FindAncestry(name)
+	for _, layer := range ancestry.Layers {
+		pbLayer, err := GetPbAncestryLayer(tx, layer, req.GetWithFeatures(), req.GetWithVulnerabilities())
```

Comment:
I was actually thinking about totally removing these from the proto, too. Let's just always return everything.

Reply:
hmm idk if we want to always return the vulnerabilities though

Reply:
Quay almost exclusively requests with vulnerabilities. IIRC the only scenario it doesn't is in tests. 
Clients that don't care about the vulnerabilities can just ignore them and this avoids us maintaining this branch needlessly.